addons/onedrive/models.py

```diff
@@ -284,3 +284,16 @@ def after_delete(self, user):
     def on_delete(self):
         self.deauthorize(add_log=False)
         self.save()
+
+    def set_auth(self, external_account, user, **kwargs):
+        super(NodeSettings, self).set_auth(external_account, user, **kwargs)
+
+        try:
+            access_token = self.fetch_access_token()
+        except exceptions.InvalidAuthError:
+            raise HTTPError(403)
+
+        client = OneDriveClient(access_token)
+        user_info = client.user_info()
+        self.drive_id = user_info['drive_id']
+        self.save()
```

Comment:
Is there a unit test that coverers this piece of `set_auth()` directly since this is one of the key part of how we store add-on use settings. I am not sure that `test_user_info_token()` is enough.

Reply:
Great point!  Most of the addon testing is done in inherited base classes, so I think I overlooked this when writing tests.  Will add.